File: scripts/extract_zips.py
import zipfile
import os

zip_files = [
    '/vercel/share/v0-project/b_9D6FyI0OfnS.zip',
    '/vercel/share/v0-project/b_Ozo4PDc7dJY.zip'
]

for zip_path in zip_files:
    print(f"\n{'='*50}")
    print(f"ZIP FILE: {os.path.basename(zip_path)}")
    print('='*50)
    
    try:
        with zipfile.ZipFile(zip_path, 'r') as zip_ref:
            # List all files in the zip
            print("\nFiles in archive:")
            for info in zip_ref.infolist():
                print(f"  - {info.filename} ({info.file_size} bytes)")
            
            # Extract and read text files
            print("\nFile contents:")
            for info in zip_ref.infolist():
                if not info.is_dir():
                    try:
                        content = zip_ref.read(info.filename)
                        # Try to decode as text
                        try:
                            text_content = content.decode('utf-8')
                            print(f"\n--- {info.filename} ---")
                            print(text_content[:5000])  # Limit output
                            if len(text_content) > 5000:
                                print(f"\n... (truncated, total {len(text_content)} chars)")
                        except UnicodeDecodeError:
                            print(f"\n--- {info.filename} --- (binary file, {len(content)} bytes)")
                    except Exception as e:
                        print(f"Error reading {info.filename}: {e}")
    except Exception as e:
        print(f"Error opening zip: {e}")

File: scripts/extract_zips.mjs
import { createReadStream } from 'fs';
import { readdir } from 'fs/promises';
import { join, basename } from 'path';
import { createUnzip } from 'zlib';
import { Parse } from 'unzipper';

const projectDir = '/vercel/share/v0-project';

async function extractZips() {
  const files = await readdir(projectDir);
  const zipFiles = files.filter(f => f.endsWith('.zip'));
  
  console.log(`Found ${zipFiles.length} zip files:\n`);
  
  for (const zipFile of zipFiles) {
    const zipPath = join(projectDir, zipFile);
    console.log(`\n${'='.repeat(50)}`);
    console.log(`ZIP FILE: ${zipFile}`);
    console.log('='.repeat(50));
    
    try {
      const entries = [];
      
      await new Promise((resolve, reject) => {
        createReadStream(zipPath)
          .pipe(Parse())
          .on('entry', async (entry) => {
            const fileName = entry.path;
            const type = entry.type;
            const size = entry.vars.uncompressedSize || 0;
            
            if (type === 'File') {
              const chunks = [];
              entry.on('data', chunk => chunks.push(chunk));
              entry.on('end', () => {
                const content = Buffer.concat(chunks);
                entries.push({ fileName, size, content });
              });
            } else {
              entry.autodrain();
            }
          })
          .on('close', resolve)
          .on('error', reject);
      });
      
      console.log('\nFiles in archive:');
      for (const entry of entries) {
        console.log(`  - ${entry.fileName} (${entry.size} bytes)`);
      }
      
      console.log('\nFile contents:');
      for (const entry of entries) {
        try {
          const text = entry.content.toString('utf-8');
          console.log(`\n--- ${entry.fileName} ---`);
          if (text.length > 5000) {
            console.log(text.slice(0, 5000));
            console.log(`\n... (truncated, total ${text.length} chars)`);
          } else {
            console.log(text);
          }
        } catch (e) {
          console.log(`\n--- ${entry.fileName} --- (binary file, ${entry.content.length} bytes)`);
        }
      }
    } catch (err) {
      console.error(`Error processing ${zipFile}:`, err.message);
    }
  }
}

extractZips();
